kge/job/train.py

```diff
@@ -672,15 +706,28 @@ def _process_batch(self, batch_index, batch) -> TrainingJob._ProcessBatchResult:
         if len(po_indexes) > 0:
             forward_time = -time.time()
             scores_po = self.model.score_po(
-                sp_po_batch[po_indexes, 0], sp_po_batch[po_indexes, 1]
+                sp_po_so_batch[po_indexes, 0], sp_po_so_batch[po_indexes, 1]
             )
-            loss_value_po = self.loss(scores_po, labels[po_indexes,]) / batch_size
+            loss_value_po = self.loss(scores_po, labels_po) / batch_size
             loss_value += loss_value_po.item()
             forward_time += time.time()
             backward_time = -time.time()
             loss_value_po.backward()
             backward_time += time.time()
 
+        # forward/backward pass (so)
```

Comment:
That's lots of repeated code. Why not loop over ```["sp_", "s_o", "_po"]``` and use an if-elif-else for computing scores/loss?